fundrebalancer/transfermonitor.go

```diff
@@ -75,7 +76,19 @@ func (t *TransferTracker) UpdateTransfers(ctx context.Context) error {
 	return nil
 }
 
-func (t *TransferTracker) updateTransferStatus(ctx context.Context, transferID int64, hash string, sourceChainID, destinationChainID string) error {
+func (t *TransferTracker) updateTransferStatus(ctx context.Context, transferID int64, createdAt time.Time, hash, sourceChainID, destinationChainID string) error {
+	rebalanceTransferTimeout := config.GetConfigReader(ctx).GetRebalanceTransferTimeout(sourceChainID)
+	if time.Since(createdAt) > *rebalanceTransferTimeout {
+		err := t.database.UpdateTransferStatus(ctx, genDB.UpdateTransferStatusParams{
+			Status: db.RebalanceTransferStatusAbandoned,
+			ID:     transferID,
+		})
+		if err != nil {
+			return fmt.Errorf("updating transfer status to abandoned for hash %s on chain %s: %w", hash, sourceChainID, err)
+		}
+		return nil
+	}
+
 	txHash, err := t.skipgo.TrackTx(ctx, hash, sourceChainID)
```

Comment:
Can we execute the abandonment logic only after we've seen a not found error from the `TrackTx` call? So it would be great to have `TrackTx` catch the not found error and return a specific `TxNotFound` (or some other naming) error type that we conditionally execute your abandonment logic above on.